xwiki-platform-core/xwiki-platform-resource/xwiki-platform-resource-servlet/src/main/java/org/xwiki/resource/servlet/AbstractServletResourceReferenceHandler.java

```diff
@@ -80,16 +80,18 @@ public void handle(ResourceReference resourceReference, ResourceReferenceHandler
                 getResourceName(typedResourceReference));
         } else if (!shouldBrowserUseCachedContent(typedResourceReference)) {
             // If we get here then either the resource is not cached by the browser or the resource is dynamic.
-            InputStream resourceStream = getResourceStream(typedResourceReference);
-            if (resourceStream != null) {
-                try {
-                    serveResource(typedResourceReference, filterResource(typedResourceReference, resourceStream));
-                } catch (ResourceReferenceHandlerException e) {
-                    this.logger.error(e.getMessage(), e);
-                    sendError(HttpStatus.SC_INTERNAL_SERVER_ERROR, e.getMessage());
+            try (InputStream resourceStream = getResourceStream(typedResourceReference)) {
+                if (resourceStream != null) {
```

Comment:
Not sure try/resource likes potentially null AutoCloseable.

Reply:
afair `try-with-resource` is `null` safe but I'll double check tomorrow.

Reply:
Indeed, you are right, just checked. I just assumed it was less clever :)